integrations/gmail/src/google-api/jwt-validation.ts

```diff
@@ -0,0 +1,89 @@
+import { google } from 'googleapis'
+import { MS_IN_12_HOURS } from 'src/utils/datetime-utils'
+import { GoogleOAuth2Client, GoogleCerts, LoginTicketPayload } from './types'
+import * as bp from '.botpress'
+
+const CERT_CACHE_STATE_NAME = 'googlePublicCertCache'
+const ALLOWED_ISSUERS = ['https://accounts.google.com']
+const REQUIRED_AUDIENCE = bp.secrets.WEBHOOK_SHARED_SECRET
+const REQUIRED_RECIPIENT = bp.secrets.WEBHOOK_SERVICE_ACCOUNT
+
+export class JWTVerifier {
+  private readonly _oauth2Client: GoogleOAuth2Client
+  private readonly _client: bp.Client
+  private readonly _ctx: bp.Context
+
+  public constructor({ ctx, client }: { ctx: bp.Context; client: bp.Client }) {
+    this._oauth2Client = new google.auth.OAuth2()
+    this._client = client
+    this._ctx = ctx
+  }
+
+  public async isJWTProperlySigned(token: string): Promise<boolean> {
+    try {
+      return await this._verifyJWTSignature(token)
+    } catch (thrown: unknown) {
+      console.error("Couldn't verify JWT signature", thrown)
```

Comment:
**question:** should this log be sent to bot builders ?

Reply:
For now no, because we only support our own OAuth GCP app, and there's nothing bot builders can do to fix this. The `console.error` ensure we log the error in the integration logs though.

In the near future I plan to add support for custom OAuth apps, like it's currently possible in the GitHub integration.